### PR TITLE
[Tests] skip cba test for unsupported platforms

### PR DESCRIPTION
**Pull request by bghimireamd**
- Added `navi48(gfx120X)` in [gtest gpu device list.](https://github.com/ROCm/MIOpen/blob/388ec314ea304aa68a350f515c28f212b6ed7035/test/gtest/gtest_common.hpp#L67)
 
- Test HW compatibility in `SetUp`

### CODE REVIEW

File: test/gtest/cba_find2_infer.cpp [hunk 1]
```diff
@@ -162,7 +162,11 @@ TEST_P(GPU_ConvBiasActivFind2InferFusionFind_FP32, ConvBiasActivFind2Float_testF
         {miopenTensorActivationY, out_dev.get()},
         {miopenTensorBias, bias_dev.get()},
     };
-
+    if(solutions.empty())
```
Comment (junliume) [inline test/gtest/cba_find2_infer.cpp:165]: This is skipping all "unsupported" platforms, not just gfx12, right?

**Review by junliume — APPROVED** [review]

**Review by CAHEK7 — COMMENTED** [review]
Since that is urgent, probably that workaround is acceptable, but proper fix is either adjusting configs for unsupported platforms (if some configs are supported and some are not) or, if the algorithm itself is unsupported, then check HW compatibility in `SetUp` and skip the test there.
Also if it's a workaround, I'd suggest making it explicit and wrapping it into ifdef like 'WORKAROUND_FOR_3224'

I can't approve this PR since there is a better solution, but I don't block it either if it's really urgent.

If it will be merged, we should create a high-priority ticket for a proper fix.

**Comment by bghimireamd** [comment]
@CAHEK7 you mean directly check inside this [ `SetUp()`](https://github.com/ROCm/MIOpen/blob/develop/test/gtest/cba_find2.hpp#L76) function? For this we need a white list of gpu's that supports cba (fusion). Then if current gpu is not in that white list then skip?

File: test/gtest/cba_find2_infer.cpp (hunk 1 above)
Comment (bghimireamd) [reply test/gtest/cba_find2_infer.cpp:165]: In a direct way yes. If `solutions` is empty then the platforms does not support cba.

**Comment by CAHEK7** [comment]
> @CAHEK7 you mean directly check inside this [ `SetUp()`](https://github.com/ROCm/MIOpen/blob/develop/test/gtest/cba_find2.hpp#L76) function? For this we need a white list of gpu's that supports cba (fusion). Then if current gpu is not in that white list then skip?

Like we do in the other cases: https://github.com/ROCm/MIOpen/blob/774b1b86635aa412fbafdeecb2a201a032861b9e/test/gtest/smoke_solver_ConvOclBwdWrW1x1.cpp#L51

File: test/gtest/cba_find2_infer.cpp [hunk 2]
```diff
@@ -162,7 +162,11 @@ TEST_P(GPU_ConvBiasActivFind2InferFusionFind_FP32, ConvBiasActivFind2Float_testF
...
+    {
+        test_skipped = true;
+        GTEST_SKIP() << "No solutions found!";
+    }
```
Comment (averinevg) [inline test/gtest/cba_find2_infer.cpp:169]: This is a bad thing, it breaks the whole test. Empty `solutions` should cause the whole test to fail. For now, let's add a device check at the beginning of this function and create a ticket to refactor this test.

**Review by averinevg — CHANGES_REQUESTED** [review]

**Comment by averinevg** [comment]
@bghimireamd Please change the description

File: test/gtest/cba_find2_infer.cpp [hunk 3]
```diff
@@ -40,6 +41,14 @@
 
 namespace cba_find2_infer {
 
+bool IsTestSupportedForDevice()
+{
+    using e_mask = enabled<Gpu::Default>;
+    //# GFX1201_DISABLED is due to WORKAROUND_SWDEV_479810
+    using d_mask = disabled<Gpu::gfx120X>;
```
Comment (averinevg) [inline test/gtest/cba_find2_infer.cpp:48]:
```suggestion
    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
    // gfx120X is not enabled due to WORKAROUND_SWDEV_479810
    using d_mask = disabled<Gpu::None>;
```

File: test/gtest/fusion_test.cpp [hunk 4]
```diff
@@ -24,13 +24,22 @@
...
 #include "get_handle.hpp"
 #include "cba.hpp"
 
 #if MIOPEN_BACKEND_HIP
+namespace fusion_test {
+bool IsTestSupportedForDevice()
+{
+    using e_mask = enabled<Gpu::Default>;
+    //# GFX1201_DISABLED is due to WORKAROUND_SWDEV_479810
+    using d_mask = disabled<Gpu::gfx120X>;
```
Comment (averinevg) [inline test/gtest/fusion_test.cpp:40]:
```suggestion
    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
    // gfx120X is not enabled due to WORKAROUND_SWDEV_479810
    using d_mask = disabled<Gpu::None>;
```

File: test/gtest/fusion_test.cpp [hunk 5]
```diff
@@ -24,13 +24,22 @@
...
 #include <gtest/gtest.h>
+#include <gtest/gtest_common.hpp>
 #include <miopen/miopen.h>
 
 #include "tensor_holder.hpp"
 #include "get_handle.hpp"
 #include "cba.hpp"
 
 #if MIOPEN_BACKEND_HIP
+namespace fusion_test {
```
Comment (averinevg) [inline test/gtest/fusion_test.cpp:35]:
```suggestion
namespace {
```

File: test/gtest/fusion_test.cpp [hunk 6]
```diff
@@ -62,6 +71,9 @@ struct GPU_FusionSetArg_FP16 : FusionSetArgTest<float>
 
 bool SkipTest() { return get_handle_xnack(); }
 
+} // namespace fusion_test
+using namespace fusion_test;
```
Comment (averinevg) [inline test/gtest/fusion_test.cpp:75]:
```suggestion
} // namespace

using GPU_FusionSetArg_FP16 = FusionSetArgTest<float>;
```

File: test/gtest/cba_find2_infer.cpp [hunk 7]
```diff
@@ -148,6 +157,11 @@ TEST_P(GPU_ConvBiasActivFind2InferFusionFind_FP32, ConvBiasActivFind2Float_testF
         test_skipped = true;
         GTEST_SKIP() << "Fusion does not support xnack";
     }
+    if(!IsTestSupportedForDevice())
+    {
+        test_skipped = true;
+        GTEST_SKIP() << "Fusion not supported in this device";
```
Comment (averinevg) [inline test/gtest/cba_find2_infer.cpp:163]:
```suggestion
        GTEST_SKIP() << "Fusion not supported for this device";
```

File: test/gtest/fusion_test.cpp [hunk 8]
```diff
@@ -72,6 +84,11 @@ TEST_P(GPU_FusionSetArg_FP16, TestSetArgApiCall)
         test_skipped = true;
         GTEST_SKIP() << "Fusion does not support xnack";
     }
+    if(!IsTestSupportedForDevice())
+    {
+        test_skipped = true;
+        GTEST_SKIP() << "CBA fusin_test not supported in this device";
```
Comment (averinevg) [inline test/gtest/fusion_test.cpp:90]:
```suggestion
        GTEST_SKIP() << "CBA fusion_test is not supported for this device";
```

File: test/gtest/graphapi_conv_bias_res_add_activ_fwd.cpp [hunk 9]
```diff
@@ -42,9 +43,16 @@
...
 
 namespace gr = miopen::graphapi;
-
 namespace conv_graph_api_test {
 
+bool IsTestSupportedForDevice()
+{
+    using e_mask = enabled<Gpu::Default>;
+    //# GFX1201_DISABLED is due to WORKAROUND_SWDEV_479810
+    using d_mask = disabled<Gpu::gfx120X>;
```
Comment (averinevg) [inline test/gtest/graphapi_conv_bias_res_add_activ_fwd.cpp:52]:
```suggestion
    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
    // gfx120X is not enabled due to WORKAROUND_SWDEV_479810
    using d_mask = disabled<Gpu::None>;
```

File: test/gtest/graphapi_conv_bias_res_add_activ_fwd.cpp [hunk 10]
```diff
@@ -213,6 +221,10 @@ class GPU_ConvBiasResAddActivation_fwd
 
     void Run()
     {
+        if(!IsTestSupportedForDevice())
+        {
+            GTEST_SKIP() << "CBA graph Fusion not supported in this device";
+        }
```
Comment (averinevg) [inline test/gtest/graphapi_conv_bias_res_add_activ_fwd.cpp:227]: Move this check to SetUp()

File: test/gtest/gtest_common.cpp [hunk 11]
```diff
@@ -99,6 +101,7 @@ const std::multimap<Gpu, DevDescription>& GetAllKnownDevices()
         {Gpu::gfx110X, {"gfx1101", 27}},
         {Gpu::gfx110X, {"gfx1101", 30}},
         {Gpu::gfx110X, {"gfx1102", 16}},
+        {Gpu::gfx120X, {"gfx1201", 0}}, // 0 is dummy value. Will replace this with real value in future.
```
Comment (averinevg) [inline test/gtest/gtest_common.cpp:104]:
```suggestion
        {Gpu::gfx120X, {"gfx1201", 10000}}, // \todo 10000 is a dummy value, replace with real value.
```

File: test/gtest/unit_conv_solver_GemmBwd1x1_stride1.cpp [hunk 12]
```diff
@@ -42,7 +42,7 @@ auto GetConvTestCases(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmBwd1x1_stride1.cpp:45]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmBwd1x1_stride2.cpp [hunk 13]
```diff
@@ -42,7 +42,7 @@ auto GetConvTestCases(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmBwd1x1_stride2.cpp:45]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmBwdRest.cpp [hunk 14]
```diff
@@ -59,7 +59,7 @@ auto GetConvTestCasesFull(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmBwdRest.cpp:62]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmFwd1x1_0_1.cpp [hunk 15]
```diff
@@ -42,7 +42,7 @@ auto GetConvTestCases(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmFwd1x1_0_1.cpp:45]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmFwd1x1_0_1_int8.cpp [hunk 16]
```diff
@@ -46,7 +46,7 @@ auto GetConvTestCases(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmFwd1x1_0_1_int8.cpp:49]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmFwd1x1_0_2.cpp [hunk 17]
```diff
@@ -42,7 +42,7 @@ auto GetConvTestCases(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmFwd1x1_0_2.cpp:45]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmFwdRest.cpp [hunk 18]
```diff
@@ -62,7 +62,7 @@ auto GetConvTestCasesFull(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmFwdRest.cpp:65]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmWrw1x1_stride1.cpp [hunk 19]
```diff
@@ -42,7 +42,7 @@ auto GetConvTestCases(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmWrw1x1_stride1.cpp:45]:
```
return Gpu::All;
```

File: test/gtest/unit_conv_solver_GemmWrwUniversal.cpp [hunk 20]
```diff
@@ -42,7 +42,7 @@ auto GetConvTestCases(miopenDataType_t datatype)
 Gpu GetSupportedDevices()
 {
     return Gpu::gfx900 | Gpu::gfx906 | Gpu::gfx908 | Gpu::gfx90A | Gpu::gfx94X | Gpu::gfx103X |
-           Gpu::gfx110X;
+           Gpu::gfx110X | Gpu::gfx120X;
```
Comment (averinevg) [inline test/gtest/unit_conv_solver_GemmWrwUniversal.cpp:45]:
```
return Gpu::All;
```

**Review by averinevg — CHANGES_REQUESTED** [review]

**Review by averinevg — APPROVED** [review]

**Review by CAHEK7 — APPROVED** [review]

**Pull request merged**